src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+        final JavaRDD<Iterable<GATKRead>> groupedReads = PSScoreUtils.groupReadsIntoPairs(pairedReads,
+                unpairedReads, scoreArgs.readsPerPartition);
+
+        //Load taxonomy database, created by running PathSeqBuildReferenceTaxonomy with this reference
+        final PSTaxonomyDatabase taxDB = PSScoreUtils.readTaxonomyDatabase(scoreArgs.taxonomyDatabasePath);
+        final Broadcast<Map<String, String>> accessionToTaxIdBroadcast = ctx.broadcast(taxDB.accessionToTaxId);
+
+        //Check header against database
+        if (scoreArgs.headerWarningFile != null) {
+            PSScoreUtils.writeHeaderWarnings(scoreArgs.headerWarningFile, header, taxDB, logger);
+        }
+
+        //Determine which alignments are valid hits and return their tax IDs in PSPathogenAlignmentHit
+        //Also adds pathseq tags containing the hit IDs to the reads
+        final JavaRDD<Tuple2<Iterable<GATKRead>, PSPathogenAlignmentHit>> readHits = PSScoreUtils.mapGroupedReadsToTax(groupedReads,
+                scoreArgs.minCoverage, scoreArgs.minIdentity, accessionToTaxIdBroadcast);
+
+        //Collect PSPathogenAlignmentHit objects
+        final Iterable<PSPathogenAlignmentHit> hitInfo = PSScoreUtils.collectTupleSecond(readHits);
+
+        //Get the original reads, now with their pathseq hit tags set
+        final JavaRDD<GATKRead> readsFinal = PSScoreUtils.flattenTupleFirstIterable(readHits);
+
+        //Compute taxonomic scores
+        final Map<String, PSPathogenTaxonScore> taxScores = PSScoreUtils.computeTaxScores(hitInfo, taxDB.tree);
+
+        //Write scores to file
+        PSScoreUtils.writeScoresFile(taxScores, taxDB.tree, scoreArgs.scoresPath);
+
+        return readsFinal;
+    }
+
+    /**
+     * Collects the second elements of all tuples in an RDD
+     */
+    public static <T, C> Iterable<C> collectTupleSecond(final JavaRDD<Tuple2<T, C>> tupleRdd) {
+        return tupleRdd.map(tuple -> tuple._2).collect();
+    }
+
+    /**
+     * Flattens RDD of tuples, in which the first elements are Iterable, to an RDD of the items in the Iterables
+     */
+    public static <T, C> JavaRDD<T> flattenTupleFirstIterable(final JavaRDD<Tuple2<Iterable<T>, C>> tupleRdd) {
+        return tupleRdd.flatMap(tuple -> tuple._1.iterator());
+    }
+
+    /**
+     * Moves reads from the same read template into an Iterable.
+     * Paired reads must be queryname-sorted, and no pair of reads can be split across partitions.
+     */
+    static JavaRDD<Iterable<GATKRead>> groupReadsIntoPairs(final JavaRDD<GATKRead> pairedReads,
+                                                           final JavaRDD<GATKRead> unpairedReads,
+                                                           final int readsPerPartitionGuess) {
+        JavaRDD<Iterable<GATKRead>> groupedReads;
+        if (pairedReads != null) {
+            groupedReads = pairedReads.mapPartitions(iter -> groupPairedReadsPartition(iter, readsPerPartitionGuess));
```

Comment:
Sometime in the past couple of months the engine changed to call putPairsInSamePartition on all inputs that are queryname sorted (which is checked for in my runTool()). This function is just grouping read pairs into Lists of size 2.